### PR TITLE
New experimental feature: Calculation of weighting potentials for undepleted detectors

### PR DESCRIPTION
**Pull request by lmh91**
In `calculate_weighting_potential!(sim, contact_id)` the keyword `depletion_handling::Bool` is now actually used.
If `depletion_handling == true` the possible undepleted region of the detector (`sim.point_types`) is used
in the application of the boundary conditions. 

The dielectric permittivity is scaled by a factor in undepleted regions. 
The factor is defined in the function `scaling_factor_for_permittivity_in_undepleted_region(::Semiconductor)`.
So the scaling can be overwritten by overwriting this internal function. 

Here, some example cases for three different scaling factors:

The upper one for an scaling factor of 1 (no scaling):

eps_1

The second one for an scaling factor of 100:

eps_100

The last one for an scaling factor of 100000 which basically mimics perfect conductivity:

eps_100000

### CODE REVIEW

**Review by fhagemann — APPROVED** [review]

File: src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetupRBCartesian3D.jl
Comment (fhagemann) [inline src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetupRBCartesian3D.jl:197]: There is a function `getindex(point_types::PointTypes, pt::AbstractCoordinatePoint)` that does exactly this [on branch v0.7](https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/v0.7/src/Types/point_types.jl#L58) (`point_types.data[find_closest_gridpoint(pt, point_types.grid)...]`). We could think about changing this once everything is merged.

Comment (lmh91) [reply src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetupRBCartesian3D.jl:197]: I will merge when the tests are through. 
Then, we could merge the current master into the v0.7 branch.

**Pull request merged**
